waku-org/nwaku
# chore(postgres): not loading the libpq library by default & better user feedback

# Description
Before this change, when a new `nwaku` user wanted to start a node, the next error appeared if the [libpq](https://www.postgresql.org/docs/9.5/libpq.html) library wasn't installed on his/her system.
```
could not load: libpq.so(.5|)
(compile with -d:nimDebugDlOpen for more information)
```

This is indeed confusing and very little information is given. Especially confusing because it appeared even the user not mounting the _Store_ protocol.

This change aims to enhance that and, by default, if the user doesn't mount the _Store_ protocol, the node will start as usual.

# Changes

- [x] Adding explicit Makefile parameter to load the _postgres_ module, and therefore, add the dependency with the libpq library.
- [x] Removing unused imports from `waku/waku_archive/archive.nim` (not directly related to the issue being enhanced in this PR.)
- [x] Adding more feedback if the user tries to mount the _Store_ protocol and use _Postgres_ as the archive backend.

## How to test

1. Run the `wakunode2` without _Store_ mounted. The node should start properly, even if the `libpq` is not being installed.
2. Run the `wakunode2` mounting _Store_ with _SQLite_. The node should start properly.
3. Run `wakunode2` mounting _Store_ with _Postgres_ and no `libpq` is present in the system. In this case, the next message should appear and the `wakunode2` ends immediately:
 "Postgres has been configured but not been compiled. Check compiler definitions."
4.
  i. `make POSTGRES=1 wakunode2 -j11`   ( In this case, the user has _postgres_ in his/her mindset )
  ii. Run `wakunode2` mounting _Store_ with _Postgres_ and no `libpq` is present in the system.
  iii. The next message should appear, which is a little bit more descriptive than the one that appeared in all the cases:
     `libpq.so.5: cannot open shared object file: No such file or directory`
     `libpq.so: cannot open shared object file: No such file or directory`
     `could not load: libpq.so(.5|)`

5. Follow all the points from 4. but having installed the `libpq`. The node should start properly.
